client/app/scripts/superdesk-vocabularies/vocabularies.js

```diff
@@ -0,0 +1,169 @@
+ * AUTHORS and LICENSE files distributed with this source code, or
+ * at https://www.sourcefabric.org/superdesk/license
+ */
+
+(function() {
+    'use strict';
+
+    VocabularyService
+        .$inject = [ 'api', 'urls', 'session', '$q'];
+    function VocabularyService(api, urls, session, $q) {
+        var service = this;
+
+        /**
+         * Fetches and caches vocabularies or returns them from the cache.
+         *
+         * @returns {Promise}
+         */
+        this.getVocabularies = function() {
+            if (typeof service.vocabularies === 'undefined') {
+                return api.query('vocabularies')
+                .then(function(result) {
+                    service.vocabularies = result;
+                    return service.vocabularies;
+                });
+            } else {
+                return $q.when(service.vocabularies);
+            }
+        };
+    }
+
+    VocabularyConfigController.$inject = ['$scope', 'vocabularies'];
+    function VocabularyConfigController($scope, vocabularies) {
+
+        /**
+         * Opens vocabulary in the edit modal.
+         *
+         * @param {Object} vocabulary
+         */
+        $scope.openVocabulary = function(vocabulary) {
+            $scope.vocabulary = vocabulary;
+        };
+
+        // load the list of vocabularies into component:
+        vocabularies.getVocabularies().then(function(vocabularies) {
+            $scope.vocabularies = vocabularies;
+        });
+    }
+
+    VocabularyEditController.$inject = [
+      '$scope',
+      'gettext',
+      'notify',
+      'api',
+      'vocabularies',
+      'metadata'
+    ];
+    function VocabularyEditController($scope, gettext, notify, api, vocabularies, metadata) {
+
+        var origVocabularyItems = _.cloneDeep($scope.vocabulary.items);
+
+        /**
+         * Unload vocabulary/close modal.
+         */
+        function closeVocabulary() {
+            $scope.vocabulary = null;
+        }
+
+        function onSuccess(result) {
+            notify.success(gettext('Vocabulary saved succesfully'));
+            closeVocabulary();
+            return result;
+        }
+
+        function onError(response) {
+            if (angular.isDefined(response.data._issues)) {
+                if (angular.isDefined(response.data._issues['validator exception'])) {
+                    notify.error(gettext('Error: ' +
+                                         response.data._issues['validator exception']));
+                } else {
+                    notify.error(gettext('Error. Vocabulary not saved.'));
+                }
+            }
+        }
+
+        /**
+         * Save current edit modal contents on backend.
+         */
+        $scope.save = function() {
+            $scope._errorUniqueness = false;
+            api.save('vocabularies', $scope.vocabulary).then(onSuccess, onError);
+            // discard metadata cache:
+            metadata.initialize();
```

Comment:
This won't force the metadata to reinitialize. https://github.com/superdesk/superdesk/blob/master/client/app/scripts/superdesk-authoring/metadata/metadata.js#L507